### DISCUSSION TITLE
has Bidirectional streaming been implemented for grpc-web?

### DISCUSSION THREAD

**Discussion by c-npolyak** [grpc/grpc-web#1416] [Q&A - General] [answered]
Just would like to know if the bidirectional streaming was implemented - since it was on the roadmap for 2023+.

Thanks

**Comment by sampajano** [comment, marked as answer]
@c-npolyak Thanks for checking! But no sorry we haven't started the work just yet! We're currently working on a migration to TS, and will start evaluating the bidi streaming feature following that! Thanks for your patience!

**Comment by sarah-spencer2** [comment]
Do you have a sense of when you think you'll start bidirectional streaming for the client, or how long this will take?

**Reply by sampajano** [reply]
The hope is that we'll get to evaluation later this year but I won't be able to promise on that totally!

**Comment by c-npolyak** [comment]
The interesting part is that I tried bidirectional and client streaming in C#/Avalonia running in browser via WebAssembly and it seems to work. 

I wonder why, C#/WebAssembly works in browser (via grpc-web) while javascript does not allow it. 

Seems like protoc does not generate the client proxies for the grpc client and bidirectional streaming calls. Is it correct? Then nodejs will not work for those streaming calls either.

**Reply by npolyak** [reply]
Want to correct previous message - the client and bidirectional streaming via WebAssembly DID NOT work in a browser - the client values were accumulating until the end of the 'client stream' and only then sent over to the server.

**Comment by emmveqz** [comment]
Hi @c-npolyak 
you may want to check out this tool in the meantime [grpc-web-native](https://github.com/emmveqz/grpc-web-native)
